### DISCUSSION TITLE
How do you in SKIP (android) handle delayed tasks on main (UI) thread ?

### DISCUSSION THREAD

**Discussion by tom-eager** [skiptools/skip#211] [Q&A] [answered]
How do you in SKIP (android) handle Delayed tasks on main thread ?  - iOS uses DispatchQueue.main.asyncAfter

**Comment by tom-eager** [comment]
Here is an example: 

DispatchQueue.main.asyncAfter(deadline: .now() + 4.0) {
        isAnimationActive = false
}

where isAnimationActive is a boolean state variable

**Reply by aabewhite** [reply]
Yes, just use that! We support translating that to a Kotlin dispatch

**Comment by tom-eager** [comment]
Skip is unable to determine the owning type for member 'now'. This often occurs when other issues prevent Skip from matching the surrounding API call, and it may resolve when those issues are fixed. Or add the owning type explicitly (e.g. MyType.now)

**Reply by aabewhite** [reply]
In the Xcode sidebar where it lists Package Dependencies, what version do you see for "skip-foundation"? Could you perhaps be using an older version where this wasn't yet implemented? You can update your libraries with File->Packages->Update to Latest Versions

Also please ensure that your file has "import Foundation"

**Comment by tom-eager** [comment]
I updated skip-foundation to 1.1.2 (it was 1.1.0) and added "import Foundation".  I then "Clear All Issues".  I still get the same error.

**Reply by aabewhite** [reply]
Thanks. This is odd, because we have code in our own samples that uses this, and I know other users are using it. Is there any way you can provide a complete code sample that reproduces the problem?

**Reply by aabewhite** [reply, marked as answer]
I should also mention that you can work around the immediate error by qualifying ".now()" with "DispatchTime.now()". But something is wrong if Skip isn't able to do that automatically

**Comment by tom-eager** [comment]
Thanks for your help on this.  I found that some files required "importing Foundation".  That change addressed not having to included "DispatchTime." before now.  I did have one instance which didn't like .now()  and it was because it used .millisecond().  I changed it to include the decimal value and it stopped complaining about .now.

**Reply by aabewhite** [reply]
I'm glad you got it working!

Yes, we only support a very narrow API on this one. We generally want people to use async/await, but we recognize that there are a few occasions when we all still need DispatchQueue.main.asyncAfter
